app/services/side-nav/menu.ts

```diff
@@ -0,0 +1,318 @@
+        }
+        return menuItem;
+      }),
+    };
+  }
+
+  @mutation()
+  private SET_LEGACY_VIEW() {
+    this.state.showCustomEditor = true;
+    this.state.compactView = false;
+
+    this.state.menuItems = {
+      ...this.state.menuItems,
+      [EMenuItem.Editor]: { ...this.state.menuItems[EMenuItem.Editor], isActive: true },
+      [EMenuItem.Themes]: { ...this.state.menuItems[EMenuItem.Themes], isActive: true },
+      [EMenuItem.AppStore]: { ...this.state.menuItems[EMenuItem.AppStore], isActive: true },
+      [EMenuItem.Highlighter]: { ...this.state.menuItems[EMenuItem.Highlighter], isActive: true },
+      [EMenuItem.LayoutEditor]: { ...this.state.menuItems[EMenuItem.LayoutEditor], isActive: true },
+      [EMenuItem.StudioMode]: { ...this.state.menuItems[EMenuItem.StudioMode], isActive: true },
+      [EMenuItem.ThemeAudit]: { ...this.state.menuItems[EMenuItem.ThemeAudit], isActive: true },
+    };
+
+    this.state[ENavName.TopNav] = {
+      ...this.state[ENavName.TopNav],
+      menuItems: [
+        { ...this.state.menuItems[EMenuItem.Editor], isActive: true },
+        { ...this.state.menuItems[EMenuItem.LayoutEditor], isActive: true },
+        { ...this.state.menuItems[EMenuItem.StudioMode], isActive: true },
+        { ...this.state.menuItems[EMenuItem.Themes], isActive: true },
+        { ...this.state.menuItems[EMenuItem.AppStore], isActive: true },
+        { ...this.state.menuItems[EMenuItem.Highlighter], isActive: true },
+        { ...this.state.menuItems[EMenuItem.ThemeAudit], isActive: true },
+      ],
+    };
+  }
+
+  @mutation()
+  private OPEN_CLOSE_MENU() {
+    this.state.isOpen = !this.state.isOpen;
+  }
+
+  @mutation()
+  private TOGGLE_SIDEBAR_SUBMENU() {
+    // currently only the custom editor needs to
+    // have the option to show/hide in the sidebar
+    this.state.showCustomEditor = !this.state.showCustomEditor;
+  }
+
+  @mutation()
+  private TOGGLE_MENU_ITEM(navName: ENavName, menuItemName: EMenuItem) {
+    // toggle boolean value
+    this.state[navName].menuItems.find(
+      (menuItem: IMenuItem) => menuItem.title === menuItemName,
+    ).isActive = !this.state[navName].menuItems.find(
+      (menuItem: IMenuItem) => menuItem.title === menuItemName,
+    ).isActive;
+  }
+
+  @mutation()
+  private TOGGLE_APP(appId: string) {
+    this.state.apps = this.state.apps.map(app => {
+      if (!app) return null;
+
+      if (app.id === appId) {
+        return { ...app, isActive: !app.isActive };
+      }
+
+      return app;
+    });
+  }
+
+  @mutation()
+  private SWAP_APP(newApp: IAppMenuItem, index: number) {
+    const updatedApps = this.state.apps.map((app, i) => {
+      if (i === index) return newApp;
+
+      if (!app || app?.id === newApp.id) {
+        // if the new app is already in the array, remove it
+        return null;
+      }
+
+      return app;
+    });
+    this.state.apps = updatedApps;
```

Comment:
this doesn't seem to handle the case of an existing app at index `i` being moved to where the target app used to be, unless that's intended

Reply:
Yes, after speaking further we Ashley about it, we decided swapping the two apps may be confusing. So the intended behavior is if the app is already slotted in a different index, we remove it from that index and add it in the new index.

Reply:
Done.